File: 백준/Silver/1417. 국회의원 선거/국회의원 선거.py
N = int(input())
dasom = int(input())
if N == 1:
    print(0)
    exit()
arr = [int(input()) for _ in range(N-1)]
cnt = 0
while True:
    if dasom > max(arr):
        break

    arr[arr.index(max(arr))] -= 1
    dasom += 1
    cnt+=1
print(cnt)

File: 백준/Silver/1417. 국회의원 선거/국회의원 선거.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Collections;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int dasom = Integer.parseInt(br.readLine());
        if (n == 1) {
            System.out.println(0);
            return;
        }

        ArrayList<Integer> arr = new ArrayList<>();
        for (int i = 0; i < n - 1; i++) {
            arr.add(Integer.parseInt(br.readLine()));
        }

        int cnt = 0;
        while (true) {
            int idx = arr.indexOf(Collections.max(arr));
            if (dasom > arr.get(idx)) {
                break;
            }

            arr.set(idx, arr.get(idx) - 1);
            dasom += 1;
            cnt += 1;
        }
        System.out.println(cnt);
    }
}
